### ISSUE TITLE
Latest unstable version of Neovim installed via Ubuntu PPA repository has blanked editor window

### ISSUE THREAD

**Issue by PaleAugust** [onivim/oni#2580]
**Oni Version:** 0.3.6
**Neovim Version (Linux only):** v0.3.2-dev
**Operating System:**
Ubuntu Linux 18.04 LTS
**Issue:**
Latest unstable version of Neovim installed via Ubuntu PPA repository has stopped editor window displaying as shown in screenshot.
**Expected behavior:**
Editor window to present cursor and Neovim interface.
**Actual behavior:**
Blank space where Neovim editor window should be as shown in screenshot.
**Steps to reproduce:**
Updated recent version of Neovim unstable using PPA, opened Oni.
Current version of Neovim unstable shown in screenshot.
Dowloaded 
![20180924-oni_error](https://user-images.githubusercontent.com/13577413/45980570-30953d80-c04a-11e8-8225-00ffab19e22d.png)
![20180924-nvim_unstable](https://user-images.githubusercontent.com/13577413/45980870-14de6700-c04b-11e8-9c59-f6dbf8c44bb7.png)

**Comment by justinmk** [comment]
What does `:ls!` show?

**Comment by PaleAugust** [comment]
As attached in neovim.  I can't run the command in Oni as there is nowhere for me to enter the command...
![20180924-ls](https://user-images.githubusercontent.com/13577413/45981286-63d8cc00-c04c-11e8-85d0-8fd0cdc99f15.png)

**Comment by justinmk** [comment]
Oh, probably not https://github.com/equalsraf/neovim-qt/issues/417 then.

Maybe https://github.com/neovim/neovim/pull/9024 is causing this, though it was intended to prevent this very issue.

**Comment by CrossR** [comment]
If nobody else has given this a test, I can try out some nvim versions this weekend and see if the later versions are causing this issue.

There is the ` "debug.neovimPath"` option that can be used to point to any nvim executable to test it without installing anything else. This includes a `.exe` on Windows, and AppImages on Linux.

**Comment by CrossR** [comment]
I've given this a test and confirmed it on the latest nightly - https://github.com/neovim/neovim/commit/6e146d413267de044a1f9f0bbb0290b5387e631c

I gave the PRs and Following HEAD docs a quick read, and then poked in Oni and it looks to be the way Oni was (hackily) getting notifications if neovim was blocked on startup.

This was done here:

https://github.com/onivim/oni/blob/1987c9fb06939e816526275731fbc160ab27c5c8/browser/src/neovim/NeovimInstance.ts#L707-L727

I'm assuming the mode check fails due to their not being a buffer at first.

Looks like we need to remove that, and to surface the errors in a different way, before we then call `nvim_ui_attach`. We could possibly do the work around specified in the PRs to preserve the old behaviour until this is implemented, so that Oni is usable for people who are following HEAD.

Any input @bryphe / @Akin909 ?

**Comment by justinmk** [comment]
> Looks like we need to remove that, and to surface the errors in a different way, before we then call `nvim_ui_attach`

I don't think so. Rather, don't surface the errors at all, let Nvim do it. The purpose of https://github.com/neovim/neovim/pull/9024 is to fix the issue in Nvim so that UIs don't need to do things like `"nvim_command_output", [":messages"]`.

> We could possibly do the work around specified in the PRs to preserve the old behaviour until this is implemented, so that Oni is usable for people who are following HEAD.

What workaround are you referring to? I think no workaround is needed with Nvim HEAD.

**Comment by justinmk** [comment]
> That would be passing both `--embed --headless`

I don't follow. If Oni is not already using `--headless` then there's no reason to add it. `--headless` is for non-UI clients.

> That said, I guess we can just get the nvim version and only run the hacky code for older versions, and have HEAD not run it and let nvim surface everything.

Yes, exactly.  That might require invoking `nvim` an extra time to get the version, then doing the "real startup". But that only costs a few milliseconds, and is nothing compared to Oni's current startup time. And it's temporary, until Oni bumps its "minimum required Nvim version".
